tanner/tests/test_api.py

```diff
@@ -1,349 +1,303 @@
+        )
 
-    def test_apply_filter_start_time_false(self):
-        filter_name = 'start_time'
-        filter_value = 148560
-
-        session = {
-            'start_time': 148555
-        }
-
-        self.returned_content = self.handler.apply_filter(filter_name, filter_value, session)
-        self.assertFalse(self.returned_content)
+        self.returned_content = self.handler.apply_filters(filters)
+        self.assertEqual(self.returned_content, self.expected_content)
 
     def test_apply_filter_end_time(self):
-        filter_name = 'end_time'
-        filter_value = 148580
-
-        session = {
-            'end_time': 148565
+        filters = {
+            "sensor_id": "9f7d7dd3-ac6b-468b-8cee-ce3e352eff6e",
+            "end_time": "11-05-2020",
         }
 
-        self.returned_content = self.handler.apply_filter(filter_name, filter_value, session)
-        self.assertTrue(self.returned_content)
-
-    def test_apply_filter_end_time_false(self):
-        filter_name = 'end_time'
-        filter_value = 148580
-
-        session = {
-            'end_time': 148590
-        }
+        self.expected_content = (
+            "SELECT S.id FROM sessions S WHERE "
+            "S.sensor_id='9f7d7dd3-ac6b-468b-8cee-ce3e352eff6e' "
+            "AND S.end_time<='2020-05-11 00:00:00'"
+        )
 
-        self.returned_content = self.handler.apply_filter(filter_name, filter_value, session)
-        self.assertFalse(self.returned_content)
+        self.returned_content = self.handler.apply_filters(filters)
+        self.assertEqual(self.returned_content, self.expected_content)
 
     def tearDown(self):
-
         async def close():
-            await self.redis_client.flushall()
-            self.redis_client.close()
-            await self.redis_client.wait_closed()
+            async with self.pg_client.acquire() as conn:
+                await conn.execute("DROP TABLE cookies;")
+                await conn.execute("DROP TABLE owners;")
+                await conn.execute("DROP TABLE paths;")
+                await conn.execute("DROP TABLE sessions;")
+
+            async with create_engine(
+                user="postgres", host="127.0.0.1", password="postgres"
+            ) as engine:
+
+                async with engine.acquire() as conn:
+                    await conn.execute(
+                        "REVOKE CONNECT ON DATABASE tanner_test_db FROM public;"
+                    )
+                    await conn.execute(
+                        """SELECT pg_terminate_backend(pg_stat_activity.pid)
+                    FROM pg_stat_activity
+                        WHERE pg_stat_activity.datname = 'tanner_test_db';
+                    """
+                    )
+                    await conn.execute("DROP database tanner_test_db")
 
         self.loop.run_until_complete(close())
+
+    @pytest.fixture(scope="session")
+    async def close_connection(self):
+        await self.pg_client.close()
```

Comment:
close should not be a coroutine according to docs

Reply:
you can also try add https://aiopg.readthedocs.io/en/stable/core.html#aiopg.Pool.wait_closed